### ISSUE TITLE
SQLAlchemy 2 compatibility issue with JupyterHub<3.0

### ISSUE THREAD

**Issue by Sheila-nk** [jupyterhub/dockerspawner#477]
### Bug description
`JupyterHub` backported the `sqalchemy` compatibility fix to JupyterHub>=3.0 but didn't restrict the `sqlalchemy` version to < 2.0 for JupyterHub<3.0. Tests on GitHub Actions are failing due to this. More info: https://github.com/jupyterhub/jupyterhub/issues/4312

![sqlalchemy error](https://user-images.githubusercontent.com/114513578/220287286-99efa1a5-db3a-40c4-b15a-d8c68caf3175.PNG)

#### Expected behaviour

Tests on GHA should pass or fail without any error messages related to sqlalchemy

#### Actual behaviour

See bug description

### How to reproduce

Making a pull request or a draft pull request triggers GHA to run, exposing the failing tests due to this bug.

### Your personal set up

 - OS:
 
 - Version(s):
 

<details><summary>Full environment</summary>

```
# paste output of `pip freeze` or `conda list` here
```
</details>

<details><summary>Configuration</summary>

```python
# jupyterhub_config.py
```
</details>

<details><summary>Logs</summary>

```
# paste relevant logs here, if any
```
</details>

**Comment by GeorgianaElena** [comment]
Thanks for opening this issue @Sheila-nk!

My proposal would be to **not frame this issue like a bug report**, but rather a proposal for discussion on the best solution forward:
- dropping support for older JupyterHub versions (see also @manics's comment in https://github.com/jupyterhub/dockerspawner/pull/476#issuecomment-1436708669)
or
- pinning sqalchemy in older JupyterHub's versions, or in dockerspawner

Also, this is not a bug necessarily since jupyterhub >= 3 has the compatibility issue fixed.

**Comment by Sheila-nk** [comment]
Closed in favor of #478
